### PR TITLE
Added Guidlines for PR's in the Contributing.md file.

### PR DESCRIPTION
**Pull request by AaronH88**
# Description

Added more guidelines for contributing to edge api

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Comment by de1987** [comment]
Where can I sign? **+1**

Just one addition, we should create a mindset to review often, because PRs might get stalled waiting a long time for reviews.

**Comment by AaronH88** [comment]
> Where can I sign? **+1**
> 
> Just one addition, we should create a mindset to review often, because PRs might get stalled waiting a long time for reviews.

Here is how I think it can be enforced: https://github.blog/2018-03-23-require-multiple-reviewers/

Waiting on the team to agree before enforcing anything :smile:

**Comment by de1987** [comment]
> > Where can I sign? **+1**
> > Just one addition, we should create a mindset to review often, because PRs might get stalled waiting a long time for reviews.
> 
> Here is how I think it can be enforced: https://github.blog/2018-03-23-require-multiple-reviewers/
> 
> Waiting on the team to agree before enforcing anything smile

Cool!
We might need to update `mergify` conditions too.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -81,3 +81,19 @@ Logging must follow the above logging key concepts as much as possible.
 - Be mindful of log levels. **Debug** is for insights of a route. **Info** should give enough information on a production service to answer most questions about what happened to a specific route for a specific customer. **Error** is for actual errors.
 
 Happy logging!
+
+## Guidlines for submitting a Pull Request
```
Comment (loadtheaccumulator) [inline CONTRIBUTING.md:85]: /Guidlines/Guidelines/

**Review by loadtheaccumulator — COMMENTED** [review]
This is a great start!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -81,3 +81,19 @@ Logging must follow the above logging key concepts as much as possible.
...
+
+In order to succesfully submit a PR we recommend the following guidlines:
+
+- Create your PR against main.
+- Keep your PR small, less than 200 lines of code.
+- Any new PR must have unit tests. Unit tests do not count towards the total number of lines of code.
+- Squash your commits.
```
Comment (loadtheaccumulator) [inline CONTRIBUTING.md:92]: For anyone unfamiliar with git and the concept of squashing, recommending...

"Squash your commits. Each PR should contain only one commit"

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -81,3 +81,19 @@ Logging must follow the above logging key concepts as much as possible.
 - Be mindful of log levels. **Debug** is for insights of a route. **Info** should give enough information on a production service to answer most questions about what happened to a specific route for a specific customer. **Error** is for actual errors.
 
 Happy logging!
+
+## Guidlines for submitting a Pull Request
+
+In order to succesfully submit a PR we recommend the following guidlines:
```
Comment (omaciel) [inline CONTRIBUTING.md:87]: s/guidlines/guidelines

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -81,3 +81,19 @@ Logging must follow the above logging key concepts as much as possible.
...
+
+- Create your PR against main.
+- Keep your PR small, less than 200 lines of code.
+- Any new PR must have unit tests. Unit tests do not count towards the total number of lines of code.
+- Squash your commits.
+- Ensure your commit message is clear, easy to read, and is very informative. We recommend using words like "Added", "Removed", "Refactored", "Deleted" and "Fixes" to outline the changes made.
+- Never merge your own PR.
+- We work off of a +2 system, where every reviewer can give a +1 if they are happy with the changes. 2 reviewers are required to merge a PR, but if you feel your PR is partially complicated, you can request more reviewers.
```
Comment (omaciel) [inline CONTRIBUTING.md:95]: s/partially/particularly

**Comment by omaciel** [comment]
How about adding guidelines around the name of the PRs themselves? For instance, if the change is related to a JIRA ticket, we could have it start with `THEEDGE-1234:` followed by the description, or the comment itself could add something like `Addresses|Resolves|Fixes THEEDGE-1234.` It could be helpful to generate better release logs automatically?

File: CONTRIBUTING.md (hunk 4 above)
Comment (AaronH88) [reply CONTRIBUTING.md]: Ty!

File: CONTRIBUTING.md (hunk 3 above)
Comment (AaronH88) [reply CONTRIBUTING.md]: Ty!

File: CONTRIBUTING.md (hunk 1 above)
Comment (AaronH88) [reply CONTRIBUTING.md]: Ty!

**Comment by AaronH88** [comment]
> How about adding guidelines around the name of the PRs themselves? For instance, if the change is related to a JIRA ticket, we could have it start with `THEEDGE-1234:` followed by the description, or the comment itself could add something like `Addresses|Resolves|Fixes THEEDGE-1234.` It could be helpful to generate better release logs automatically?

Good suggestion. Added, and expanded slightly to include Bugzilla.

**Comment by loadtheaccumulator** [comment]
@AaronH88 How about a technical debt line about rather than going back to change, consolidate, etc. in a refactor later, just do it then--even if it runs you past Sprint end.

**Comment by AaronH88** [comment]
> @AaronH88 How about a technical debt line about rather than going back to change, consolidate, etc. in a refactor later, just do it then--even if it runs you past Sprint end.

Interesting, can I get a sentence from you on how to word that please?

**Comment by loadtheaccumulator** [comment]
> Interesting, can I get a sentence from you on how to word that please?

"TECHNICAL DEBT: When dropping a // FIXME: or // TODO: in your code, consider going ahead and making that change--even if it means crossing into the next Sprint.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -81,3 +81,21 @@ Logging must follow the above logging key concepts as much as possible.
...
 
 Happy logging!
+
+## Guidelines for submitting a Pull Request
+
+In order to successfully submit a PR we recommend the following guidelines:
+
+- Create your PR against main.
+- Keep your PR small, less than 200 lines of code.
+- Any new PR must have unit tests. Unit tests do not count towards the total number of lines of code.
```
Comment (acosferreira) [inline CONTRIBUTING.md:91]: Can we add a coverage validation to dont allow merge if decreases?

Comment (AaronH88) [reply CONTRIBUTING.md]: Added

File: CONTRIBUTING.md (hunk 2 above)
Comment (AaronH88) [reply CONTRIBUTING.md]: Added

**Review by loadtheaccumulator — APPROVED** [review]

**Review by mgold1234 — APPROVED** [review]
I like the idea!
/lgtm

**Review by ldjebran — APPROVED** [review]

**Pull request merged**